### PR TITLE
Fix Number to Datetime conversion

### PR DESCRIPTION
**Pull request by riverszhang89**
Handle the decimal fraction of a negative number correctly, when counting from the Unix epoch.

### CODE REVIEW

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>5/401 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>simple_timepart_reptimeout_generated<br>sc_timepart<br>simple_timepart<br>datetimeusec<br>datetime</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/9c2f5d57/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/9c2f5d57/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2753'>Resubmit</a></p>

File: db/types.c [hunk 1]
```diff
@@ -13400,11 +13400,23 @@ int real_to_dttz(double d, dttz_t *dt, int precision)
     dt->dttz_conv = 1;
     dt->dttz_sec = d;
     if ((long long)(d * 1000) == d * 1000) {
-        dt->dttz_frac = (d - (long long)d) * 1E3 + 0.5;
+        dt->dttz_frac = (d - (long long)d) * 1E3;
```
Comment (adizaimi) [inline db/types.c:13403]: since we already casted for the assignment in 13401, maybe change to:
  dt->dttz_frac = (d - dt->dttz_sec) * 1E3;

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>4/401 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>simple_timepart_reptimeout_generated<br>simple_timepart<br>datetime<br>logfill</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/c27cab6d/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/c27cab6d/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2753'>Resubmit</a></p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/c27cab6d/robomark.txt'>Error &#x26A0; </a>.<br>Regression testing: <b>4/403 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>constraints_noreorder_generated<br>simple_timepart_reptimeout_generated<br>sc_timepart<br>simple_timepart</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/c27cab6d/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/c27cab6d/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2753'>Resubmit</a></p>

**Review by adizaimi — APPROVED** [review]
LGTM

**Comment by riverszhang89** [comment]
@adizaimi Thank you for the review!

**Pull request merged**